central/reports/scheduler/v2/reportgenerator/email_formatter.go

```diff
@@ -0,0 +1,204 @@
+	var writer strings.Builder
+	writer.WriteString(emailBody)
+	writer.WriteString("\n\n")
+	writer.WriteString(configDetailsHtml)
+
+	return writer.String()
+}
+
+func formatReportConfigDetails(snapshot *storage.ReportSnapshot, numDeployedImageCVEs, numWatchedImageCVEs int) (string, error) {
+	var writer strings.Builder
+
+	err := validateSnapshot(snapshot)
+	if err != nil {
+		return "", err
+	}
+	reportFilters := snapshot.GetVulnReportFilters()
+
+	writer.WriteString("<div>")
+
+	// Config name
+	formatSingleDetail(&writer, "Config name", snapshot.GetName())
+
+	// Number of CVEs found
+	formatSingleDetail(&writer, "Number of CVEs found",
+		fmt.Sprintf("%d in Deployed images", numDeployedImageCVEs),
+		fmt.Sprintf("%d in Watched images", numWatchedImageCVEs))
+
+	// Severities
+	// create a copy because severities will be sorted in descending order (critical, important, moderate, low)
+	severities := append([]storage.VulnerabilitySeverity{}, reportFilters.GetSeverities()...)
+	slices.SortFunc(severities, func(s1, s2 storage.VulnerabilitySeverity) bool {
+		return s1 > s2
+	})
+	formatSingleDetail(&writer, "CVE severity", severities...)
+
+	// Fixability
+	fixabilities := expandFixability(reportFilters.GetFixability())
+	formatSingleDetail(&writer, "CVE status", fixabilities...)
+
+	// Collection
+	formatSingleDetail(&writer, "Report scope", snapshot.GetCollection())
+
+	// Image types
+	// create a copy because image types will be sorted in ascending order (deployed, watched)
+	imageTypes := append([]storage.VulnerabilityReportFilters_ImageType{}, reportFilters.GetImageTypes()...)
+	slices.Sort(imageTypes)
+	formatSingleDetail(&writer, "Image type", imageTypes...)
+
+	// CVEs discovered since
+	formatSingleDetail(&writer, "CVEs discovered since", reportFilters.GetCvesSince())
+
+	writer.WriteString("</div>")
+
+	return writer.String(), nil
+}
+
+func expandFixability(fixability storage.VulnerabilityReportFilters_Fixability) []storage.VulnerabilityReportFilters_Fixability {
+	if fixability == storage.VulnerabilityReportFilters_BOTH {
+		return []storage.VulnerabilityReportFilters_Fixability{
+			storage.VulnerabilityReportFilters_FIXABLE,
+			storage.VulnerabilityReportFilters_NOT_FIXABLE,
+		}
+	}
+	return []storage.VulnerabilityReportFilters_Fixability{fixability}
+}
+
+func formatSingleDetail[T any](writer *strings.Builder, heading string, values ...T) {
+	writer.WriteString("<div style=\"padding: 0 0 10px 0\">")
+
+	// Add heading
+	writer.WriteString("<span style=\"font-weight: bold; margin-right: 10px\">")
+	writer.WriteString(fmt.Sprintf("%s: ", heading))
+	writer.WriteString("</span>")
+
+	// Add values
+	if len(values) > 0 {
+		writer.WriteString("<span>")
+		for i, valI := range values {
+			writer.WriteString(convertValueToFriendlyText(valI))
+			if i < (len(values) - 1) {
+				writer.WriteString(", ")
+			}
+		}
+		writer.WriteString("</span>")
+	}
+
+	writer.WriteString("</div>")
```

Comment:
Just a simple multi-line "key": "value" pair structure would remove the markup requirement altogether and would work with all notifiers.

Reply:
I think the email does look better with some markup added. But I also agree that it should be usable by other notifiers as well. In a followup PR, I would try to add a text based table (no markup) and that pattern can be used for other notifiers as well.